flintrock/services.py

```diff
@@ -202,28 +202,29 @@ def health_check(self, master_host: str):
 
 
 class Spark(FlintrockService):
-    def __init__(self, version: str=None, git_commit: str=None, git_repository: str=None):
+    def __init__(self, version: str=None, download_source: str=None,
+                 git_commit: str=None, git_repository: str=None):
         # TODO: Convert these checks into something that throws a proper exception.
         #       Perhaps reuse logic from CLI.
         assert bool(version) ^ bool(git_commit)
         if git_commit:
             assert git_repository
 
         self.version = version
+        self.download_source = download_source
         self.git_commit = git_commit
         self.git_repository = git_repository
 
         self.manifest = {
             'version': version,
+            'download_source': download_source,
             'git_commit': git_commit,
             'git_repository': git_repository}
 
     def install(
             self,
             ssh_client: paramiko.client.SSHClient,
             cluster: FlintrockCluster):
-        # TODO: Allow users to specify the Spark "distribution". (?)
-        distribution = 'hadoop2.6'
```

Comment:
As a follow up we could support a `{d}` template in `download_source` as is done with the version with `{v}`.